aviary/modules/assembly/scripts/spades_assembly.py

```diff
@@ -0,0 +1,91 @@
+import subprocess
+import os
+import sys
+from typing import List
+
+def spades_asssembly(
+    input_fastq: str,
+    input_long_reads: str,
+    output_fasta: str,
+    output_spades_folder: str,
+    max_memory: int,
+    threads: int,
+    kmer_sizes: List[int],
+    tmp_dir: str,
+    long_read_type: str,
+    log: str):
+
+    '''
+    Assemble short reads and long reads (if any) using spades
+    :param input_fastq: short reads fastq file
+    :param input_long_reads: long reads fastq file
+    :param output_fasta: output fasta file
+    :param output_spades_folder: output spades folder
+    :param max_memory: maximum memory to use
+    :param threads: number of threads
+    :param tmpdir: temporary directory
+    :param kmer_sizes: list of kmer sizes
+    :param long_read_type: type of long reads
+    :param log: log file
+    :return:
+    '''
+    if tmp_dir:
+        tmp_dir_arg = f"--tmp-dir {tmp_dir}"
+    else:
+        try:
+            tmp_dir = os.environ["TMPDIR"]
+            tmp_dir_arg = f"--tmp-dir {tmp_dir}"
+        except KeyError:
+            tmp_dir_arg = ""
+    
+    if os.path.exists("data/spades_assembly/tmp"):
+        subprocess.call("rm -rf data/spades_assembly/tmp",shell=True)
```

Comment:
We shouldn't be using `subprocess.call` with `shell=True`, it isn't safe. I know it is what snakemake does when shell commands are run, but we can do better. Refer to other python scripts on how to avoid using `shell=True`. `run` is a safe alternative, `run_flye.py` has a proper implementation